xbmc/addons/kodi-addon-dev-kit/src/api2/kodi-addon-sharedlibrary/CMakeLists.txt

```diff
@@ -17,54 +17,92 @@ set(kodi-addon-sharedlibrary_VERSION ${ADDON_API_VERSION})
+            gui/AddonLib_GUI_DialogContextMenu.cpp
+            gui/AddonLib_GUI_DialogExtendedProgress.cpp
+            gui/AddonLib_GUI_DialogFileBrowser.cpp
+            gui/AddonLib_GUI_DialogKeyboard.cpp
+            gui/AddonLib_GUI_DialogNumeric.cpp
+            gui/AddonLib_GUI_DialogOK.cpp
+            gui/AddonLib_GUI_DialogProgress.cpp
+            gui/AddonLib_GUI_DialogSelect.cpp
+            gui/AddonLib_GUI_DialogTextViewer.cpp
+            gui/AddonLib_GUI_DialogYesNo.cpp
+            gui/AddonLib_GUI_ListItem.cpp
+            gui/AddonLib_GUI_Window.cpp
+            inputstream/AddonLib_InputStream.cpp
+            peripheral/AddonLib_Peripheral.cpp
+            player/AddonLib_Player_InfoTagMusic.cpp
+            player/AddonLib_Player_InfoTagVideo.cpp
+            player/AddonLib_Player_Player.cpp
+            player/AddonLib_Player_PlayList.cpp
+            pvr/AddonLib_PVR_General.cpp
+            pvr/AddonLib_PVR_Stream.cpp
+            pvr/AddonLib_PVR_Transfer.cpp
+            pvr/AddonLib_PVR_Trigger.cpp)
+            
+set(HEADERS InterProcess.h
+            ErrorCodeNames.h
+            ../../../include/kodi/kodi_adsp_dll.h
+            ../../../include/kodi/kodi_adsp_types.h
+            ../../../include/kodi/kodi_audiodec_dll.h
+            ../../../include/kodi/kodi_audiodec_types.h
+            ../../../include/kodi/kodi_audioengine_types.h
+            ../../../include/kodi/kodi_inputstream_dll.h
+            ../../../include/kodi/kodi_inputstream_types.h
+            ../../../include/kodi/kodi_peripheral_callbacks.h
+            ../../../include/kodi/kodi_peripheral_dll.h
+            ../../../include/kodi/kodi_peripheral_types.h
+            ../../../include/kodi/kodi_peripheral_utils.hpp
+            ../../../include/kodi/kodi_vfs_types.h
+            ../../../include/kodi/kodi_vfs_utils.hpp
+            ../../../include/kodi/libKODI_adsp.h
+            ../../../include/kodi/libKODI_audioengine.h
+            ../../../include/kodi/libKODI_guilib.h
+            ../../../include/kodi/libKODI_inputstream.h
+            ../../../include/kodi/libKODI_peripheral.h
+            ../../../include/kodi/libXBMC_addon.h
+            ../../../include/kodi/libXBMC_codec.h
+            ../../../include/kodi/libXBMC_pvr.h
+            ../../../include/kodi/xbmc_addon_cpp_dll.h
+            ../../../include/kodi/xbmc_addon_dll.h
+            ../../../include/kodi/xbmc_addon_types.h
+            ../../../include/kodi/xbmc_audioenc_dll.h
+            ../../../include/kodi/xbmc_audioenc_types.h
+            ../../../include/kodi/xbmc_codec_types.h
+            ../../../include/kodi/xbmc_epg_types.h
+            ../../../include/kodi/xbmc_pvr_dll.h
+            ../../../include/kodi/xbmc_pvr_types.h
+            ../../../include/kodi/xbmc_scr_dll.h
+            ../../../include/kodi/xbmc_scr_types.h
+            ../../../include/kodi/xbmc_vis_dll.h
+            ../../../include/kodi/xbmc_vis_types.h)
```

Comment:
You can also do this changes on the V3 (xbmc/addons/kodi-addon-dev-kit/src/api3/kodi-addon-sharedlibrary/CMakeLists.txt) if not I can do it.

As info, the V3 is to use for further development. If no change is needed are in his function table (v3) the already present function addresses from level 2 added. With this is on a Level change only the new function needed to add, all other are from older level.